core/src/main/java/bio/terra/pearl/core/service/workflow/EnrollmentService.java

```diff
@@ -131,25 +136,35 @@ public HubResponse<Enrollee> enroll(String portalShortcode, EnvironmentName envN
     }
 
     /**
-     * Will create an enrollee for both the given user and a new governed user that is registered.  The HubResponse
-     * returned will include the governedEnrollee as the "response", and the enrollee corresponding to the proxy user as
-     * the "enrollee"
+     * Will check if proxy is already enrolled, if not will create an enrollee for the proxy. Will then create a new
+     * governed user that is registered.  The HubResponse returned will include the governedEnrollee as the "response",
+     * and the enrollee corresponding to the proxy user as the "enrollee"
      */
     @Transactional
-    public HubResponse<Enrollee> enrollAsProxy(String portalShortcode, EnvironmentName envName, String studyShortcode, ParticipantUser user, PortalParticipantUser ppUser, UUID preEnrollResponseId) {
-        HubResponse<Enrollee> proxyResponse = enroll(portalShortcode, envName, studyShortcode, user, ppUser, preEnrollResponseId, false);
-        HubResponse<Enrollee> governedResponse = enrollGovernedUser(portalShortcode, envName, studyShortcode, proxyResponse.getEnrollee(), user, ppUser, preEnrollResponseId);
-        governedResponse.setEnrollee(proxyResponse.getEnrollee());
+    public HubResponse<Enrollee> enrollAsProxy(EnvironmentName envName, String studyShortcode, ParticipantUser proxyUser,
+                                               PortalParticipantUser ppUser, UUID preEnrollResponseId) {
+        Optional<Enrollee> maybeProxyEnrollee =
+                enrolleeService.findOneByParticipantUserIdAndStudyEnvironmentId(proxyUser.getId(), studyShortcode, envName);
+        if (maybeProxyEnrollee.isEmpty()) {
+            maybeProxyEnrollee = Optional.of(enroll(envName, studyShortcode, proxyUser, ppUser, null, false).getEnrollee());
+        }
+        Enrollee proxyEnrollee = maybeProxyEnrollee.get();
```

Comment:
```suggestion
        Enrollee proxyEnrollee = maybeProxyEnrollee.orElseGet(() -> enroll(envName, studyShortcode, proxyUser, ppUser, null, false).getEnrollee());
```

This would allow you to get rid of the `.get()` call here

Reply:
Weirdly this actually ends up in a race condition and a PSQL duplicate error for the case of 2 enrollees with the same proxy. I _think_ what happens is they both attempt the enroll at the same time. I tried adding `@Transactional` to every method and calling them with `this` but I think we may have to change how enrollees are populated to get it fixed. I am investigating this more

Reply:
I've been doing some background reading, and it seems like the consensus is that local variables of Optional<> type are a 'code smell'.  I'm still getting used to functional-style java programming, so this is a bit of a change for me (in the past, I liked line-by-line imperative spelling out of what was going on).  But I think this is where most Java libraries and codebases are headed, so we should do our best to lean into a more functional style.  So I would change this to:
```
Enrollee proxyEnrollee = enrolleeService.findOneByParticipantUserIdAndStudyEnvironmentId(proxyUser.getId(), studyShortcode, envName)
                .orElseGet(() -> enroll(envName, studyShortcode, proxyUser, ppUser, null, false).getEnrollee());
```